Gruvo/Gruvo/Controllers/ProfileController.cs

```diff
@@ -188,6 +188,34 @@ public IActionResult GetUserTweets(long? id)
             }
         }
 
+        [Route("userTweetsBatch")]
+        [HttpPost]
+        public IActionResult GetUserTweetsBatch([FromBody] ProfileState state)
+        {
+            IEnumerable<ReadableTweet> arr = null;
+            try
+            {
+                string cookie = Request.Cookies["Gruvo"];
+                long userid = _tokenUserPairs.Pairs[cookie].Id;
+
+                if (state.id.HasValue)
+                {
+                    arr = _repository.TweetDAO.GetUserPostsBatch(state.id.Value,true, state.date);
+                    if (arr == null) throw new NullReferenceException();
```

Comment:
no no no no no nooooo
Don't throw exception where it should not be thrown. It's a good question for interview (how to make some logic without using "if" but it's awful in real projects.
Each throw - catch is 100 or more times heavier then a simple if.
Just return here BadRequest("No posts found") or smth went wrong or I don't know) but not an exception

Reply:
BTW, you will never have null in current implementation, because you set list=new list in the second line of GetUserPostsBatch